testutil/ensurelogchecker.go

```diff
@@ -0,0 +1,120 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package testutil
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"strings"
+
+	"gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+func getReceiver(funcDecl *ast.FuncDecl) (string, bool) {
+	if funcDecl.Recv != nil && len(funcDecl.Recv.List) > 0 {
+		if starExpr, ok := funcDecl.Recv.List[0].Type.(*ast.StarExpr); ok {
+			if ident, ok := starExpr.X.(*ast.Ident); ok {
+				return ident.Name, true
+			}
+		} else if ident, ok := funcDecl.Recv.List[0].Type.(*ast.Ident); ok {
+			return ident.Name, true
+		}
+	}
+	return "", false
+}
+
+func getChildEnsureList(fset *token.FileSet, fileContent string, file *ast.File) []string {
+	for _, decl := range file.Decls {
+		funcDecl, ok := decl.(*ast.FuncDecl)
+		if !ok || funcDecl.Name.Name != "Ensure" {
+			continue
+		}
+		ensures := []string{}
```

Comment:
sorry missed this one in the previous comment, this can start as nil (so we don't sometimes return `nil` and some an empty slice)
```suggestion
		var ensures []string
```